vaadin-editable-label-demo/src/main/java/software/xdev/vaadin/editable_label/example/HomeView.java

```diff
@@ -1,98 +1,221 @@
+		formLayout.setResponsiveSteps(new FormLayout.ResponsiveStep(
+			"0",
+			2,
+			FormLayout.ResponsiveStep.LabelsPosition.TOP));
+		
+		// Types are required here otherwise a compile error occurs (see https://github.com/vaadin/flow/issues/7920)
+		Map.<String, AbstractEditableLabel<?, ?, ?>>ofEntries(
+				entry(
+					"TextField",
+					new EditableLabelTextField()
+						.withValue("Some text")),
+				entry(
+					"TextField with empty value",
+					new EditableLabelTextField()),
+				entry(
+					"Textarea",
+					new EditableLabelTextArea()
+						.withValue(
+							"Lorem ipsum dolor sit amet, consectetur adipisici elit, sed eiusmod tempor incidunt ut "
+								+ "labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation "
+								+ "ullamco laboris nisi ut aliquid ex ea commodi consequat. Quis aute iure "
+								+ "reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.")),
+				entry(
+					"NumberField",
+					new EditableLabelNumberField()
+						.withValue(123.45)),
+				entry(
+					"BigDecimalField",
+					new EditableLabelBigDecimalField()
+						.withValue(BigDecimal.ONE)),
+				entry(
+					"BigDecimalField with currency",
+					new EditableLabelBigDecimalField()
+						.withValue(BigDecimal.TEN)
+						.withNumberFormat(NumberFormat.getCurrencyInstance())),
+				entry(
+					"ComboBox",
+					new EditableLabelComboBox<Vehicle>()
+						.withItems(Vehicle.values())
+						.withValue(Vehicle.PLANE)
+						.withLabelGenerator(Vehicle::getEmoji)),
+				entry(
+					"ComboBox with empty value 🚲",
+					new EditableLabelComboBox<Vehicle>(el -> el.getEditor().setClearButtonVisible(true))
+						.withItems(Vehicle.values())
+						.withLabelGenerator(Vehicle::getEmoji, "🚲")),
+				entry(
+					"DatePicker",
+					new EditableLabelDatePicker()
+				),
+				entry(
+					"DatePicker with I18N",
+					new EditableLabelDatePicker(getDatepickerWithI18N())
+						.withValue(LocalDate.of(2000, 1, 1))
+						.withTryUseI18NFormat()
+				)
+			).entrySet()
+			.stream()
+			.sorted(Map.Entry.comparingByKey())
+			.forEach(e -> {
+				this.registerValueChangeEvent(e.getKey(), e.getValue());
+				formLayout.addFormItem(e.getValue(), e.getKey());
+			});
+		
+		return formLayout;
 	}
 	
-	private Component createEditableLabelPresenter(final String name, final HasValue<?, ?> editableLabelComponent)
+	private VerticalLayout getCustomComponent()
 	{
-		final VerticalLayout vl = new VerticalLayout();
-		vl.add(new H4(name));
-		vl.add((Component)editableLabelComponent);
+		final String defaultValue = "example@example.com";
+		final EditableLabel<EmailField, String> emailLabel = new EditableLabel<>(new EmailField())
+			.withValue(defaultValue);
 		
-		this.addValueChangeListener(name, editableLabelComponent);
+		this.registerValueChangeEvent("Custom-Component", emailLabel);
 		
+		final Button btnRestoreDefault = new Button("Restore default", ev -> emailLabel.setValue(defaultValue));
```

Comment:
We should rename the buttons "Restore default" and "Restore default in 5s" to sth. like "Restore Custom Component". 
I thought at first, that those button would restore everything I've changed on all components. 
Therefore, these are not very intuitive.